pkg/cloud/retry_manager.go

```diff
@@ -0,0 +1,50 @@
+package cloud
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+)
+
+const (
+	// retryMaxAttempt sets max number of EC2 API call attempts.
+	// Set high enough to ensure default sidecar timeout will cancel context long before we stop retrying.
+	retryMaxAttempt = 50
+)
+
+// newAdaptiveRetryer restricts attempts of API calls that recently hit throttle errors.
+func newAdaptiveRetryer() *retry.AdaptiveMode {
+	return retry.NewAdaptiveMode(func(ao *retry.AdaptiveModeOptions) {
+		ao.StandardOptions = append(ao.StandardOptions, func(so *retry.StandardOptions) {
+			so.MaxAttempts = retryMaxAttempt
+		})
+	})
+}
+
+// retryManager dictates the retry strategies of EC2 API calls.
+// Each mutating EC2 API has its own retryer because the AWS SDK throttles on a retryer object level, not by API name.
+// While default AWS accounts share request tokens between mutating APIs, users can raise limits for individual APIs.
+// Separate retryers ensures that throttling one API doesn't unintentionally throttle others with separate token buckets.
+type retryManager struct {
+	createVolumeRetryer                            *retry.AdaptiveMode
```

Comment:
small np: is it possible to have a map of `<something>:*retry.AdaptiveMode`?

Reply:
Is there a benefit to using a map here? 

I went for a struct to avoid defining [extra key constants](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/64772f0ffef94daa4b03a0115126bd80c1cfb72f/pkg/driver/constants.go#L35) and for extra type-safety protection from typos. Open to changing though.

Reply:
I thought about this, but I think that all the map does is add additional (unnecessary) checks at each call site for non-nilness.

A map would be useful here if we could dynamically setup the retryer based on the API name but that's not possible.